[CS2113T-F11-2] BudgetTracker

WeiXuanYap opened the pull request:
Target User: Tech-savvy students who have trouble keeping track of their expenses.
Value Proposition: Set guidelines for our users' spending habits so they do not exceed their budget and allow them to cut down on spending on areas which they tend to overspend.

APZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,142 @@
+Value proposition: allow users to record and track expenses more conveniently,
+and thereby encourage them to have greater control over their finances.
 
 ## User Stories
```
Should the user stories be sorted by version or type of user? Would it be neater and clearer?

APZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,142 @@
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Implementation
+###Delete feature
```
Missing a space after the '###', resulting in 'Delete feature' to not be identified as a header.

APZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,142 @@
+<br/> **Step 5**. The Parser starts to parse the command, it extracts starting index 3, ending index 5, and month 10. As this is a range of expenditures to be deleted, the Parser class calls DeleteMultipleExpenditureCommand to work. By using a for loop, the 3 expenditures are successfully deleted and prints out a showMultipleExpenditureDeletedMessage from TextUi class.
+
+
+###Edit feature
```
Missing a space after the '###', resulting in 'Edit feature' to not be identified as a header.

APZH commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,142 @@
+original source as well}
 
-## Design & implementation
+## Design 
```
May I suggest the inclusion of UML diagrams to supplement the explanation of the features stated below, on top on the step by step guide currently included.

APZH reviewed (commented):
Week 11 Tutorial Review

YEOWEIHNGWHYELAB committed:
test case

EdisonZhong17 committed:
Merge pull request #208 from LS-Young/master

edit helpcommand and listrecordscommand

YEOWEIHNGWHYELAB committed:
added budget test

YEOWEIHNGWHYELAB committed:
test

YEOWEIHNGWHYELAB committed:
formatting

YEOWEIHNGWHYELAB committed:
added test case

YEOWEIHNGWHYELAB committed:
formatting

YEOWEIHNGWHYELAB committed:
test

YEOWEIHNGWHYELAB committed:
test

YEOWEIHNGWHYELAB committed:
test

YEOWEIHNGWHYELAB committed:
Merge pull request #211 from YEOWEIHNGWHYELAB/master

test case

YEOWEIHNGWHYELAB committed:
write test

YEOWEIHNGWHYELAB committed:
Merge pull request #212 from YEOWEIHNGWHYELAB/master

write test

YEOWEIHNGWHYELAB committed:
ug

YEOWEIHNGWHYELAB committed:
Merge pull request #213 from YEOWEIHNGWHYELAB/master

ug

EdisonZhong17 committed:
Update DG with implementation for Ui

EdisonZhong17 committed:
Merge pull request #214 from EdisonZhong17/master

Update DG with implementation for Ui

EdisonZhong17 committed:
EdisonZhong17 ppp update

EdisonZhong17 committed:
Merge branch 'AY2122S1-CS2113T-F11-2:master' into master

EdisonZhong17 committed:
Merge pull request #215 from EdisonZhong17/master

EdisonZhong17 ppp update

jyxhazcake committed:
Add class diagram for logic component

YEOWEIHNGWHYELAB committed:
storage test

WeiXuanYap committed:
Merge remote-tracking branch 'upstream/master'

YEOWEIHNGWHYELAB committed:
test

YEOWEIHNGWHYELAB committed:
test

jyxhazcake committed:
Add half of junit tests for parserUtil

YEOWEIHNGWHYELAB committed:
test

YEOWEIHNGWHYELAB committed:
Merge pull request #216 from YEOWEIHNGWHYELAB/master

storage test

WeiXuanYap committed:
Add Testing +DG Diagrams

1. Added JUnit test to AddExpenditureCommand and AddBudgetCommand
2. Added DG Diagrams for Data Component

WeiXuanYap committed:
Fix checkstyle

jyxhazcake committed:
Update DG Logic

jyxhazcake committed:
Merge pull request #217 from ZenoNexus/master

Add Testing +DG Diagrams

jyxhazcake committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp into DeveloperGuide

# Conflicts:
#	docs/DeveloperGuide.md

YEOWEIHNGWHYELAB committed:
puml diagram for storage part 1

YEOWEIHNGWHYELAB committed:
Merge pull request #219 from YEOWEIHNGWHYELAB/master

puml diagram for storage part 1

YEOWEIHNGWHYELAB committed:
Merge pull request #218 from jyxhazcake/DeveloperGuide

Update DG Logic component

jyxhazcake committed:
Merge branch 'master' into junit_tests

YEOWEIHNGWHYELAB committed:
minor change to reloadArray UML

YEOWEIHNGWHYELAB committed:
csv sequence diagram

YEOWEIHNGWHYELAB committed:
Merge pull request #220 from YEOWEIHNGWHYELAB/master

csv sequence diagram

YEOWEIHNGWHYELAB committed:
tag

YEOWEIHNGWHYELAB committed:
Merge pull request #221 from YEOWEIHNGWHYELAB/master

tag

YEOWEIHNGWHYELAB committed:
getting started

YEOWEIHNGWHYELAB committed:
images

YEOWEIHNGWHYELAB committed:
Merge pull request #222 from YEOWEIHNGWHYELAB/master

getting started

WeiXuanYap committed:
Fix minor bugs

jyxhazcake committed:
Add JUnit testing for ParserUtil

WeiXuanYap committed:
Fix checkstyle issues

EdisonZhong17 committed:
Merge pull request #223 from ZenoNexus/master

Fix minor bugs

EdisonZhong17 committed:
Fix bug that deleting with range of indexes make the UI look very messy

EdisonZhong17 committed:
Fix bug that deleting with range of indexes make the UI look very messy

EdisonZhong17 committed:
Fix bug that deleting with range of indexes make the UI look very messy

WeiXuanYap committed:
change theme

from merlot -> cayman

WeiXuanYap committed:
Merge pull request #225 from EdisonZhong17/master

Fix bug that deleting with range of indexes make the UI look very messy

YEOWEIHNGWHYELAB committed:
Merge pull request #226 from ZenoNexus/master

change theme

YEOWEIHNGWHYELAB committed:
dg update

YEOWEIHNGWHYELAB committed:
Merge pull request #227 from YEOWEIHNGWHYELAB/master

dg update

WeiXuanYap committed:
Fix User Guide formatting

WeiXuanYap committed:
Fix User Guide Formatting

WeiXuanYap committed:
Fix User Guide Formatting

WeiXuanYap committed:
Merge pull request #228 from ZenoNexus/master

Fix User Guide formatting

WeiXuanYap committed:
Fix User Guide Formatting

YEOWEIHNGWHYELAB committed:
datasample

YEOWEIHNGWHYELAB committed:
Merge pull request #230 from YEOWEIHNGWHYELAB/master

datasample

YEOWEIHNGWHYELAB committed:
Merge pull request #229 from ZenoNexus/master

Fix User Guide Formatting

jyxhazcake committed:
Merge branch 'junit_tests' of https://github.com/jyxhazcake/tp into junit_tests

YEOWEIHNGWHYELAB committed:
Storage

YEOWEIHNGWHYELAB committed:
Explain sequence diagram

YEOWEIHNGWHYELAB committed:
Merge pull request #231 from YEOWEIHNGWHYELAB/master

sequence diagram explanation

YEOWEIHNGWHYELAB committed:
Storage class diagram

YEOWEIHNGWHYELAB committed:
Merge pull request #232 from YEOWEIHNGWHYELAB/master

Storage class diagram

YEOWEIHNGWHYELAB committed:
Storage doc

YEOWEIHNGWHYELAB committed:
doc

YEOWEIHNGWHYELAB committed:
Merge pull request #233 from YEOWEIHNGWHYELAB/master

Storage doc

YEOWEIHNGWHYELAB committed:
fix explosion alignment

YEOWEIHNGWHYELAB committed:
Merge pull request #234 from YEOWEIHNGWHYELAB/master

fix explosion alignment

EdisonZhong17 committed:
update DG

EdisonZhong17 committed:
update DG

EdisonZhong17 committed:
update DG

EdisonZhong17 committed:
update DG

jyxhazcake committed:
Add JUnit Testing for Parser and AddLoan

jyxhazcake committed:
Merge pull request #235 from EdisonZhong17/master

Update UG

LS-Young committed:
Merge pull request #236 from jyxhazcake/junit_tests

Add JUnit Tests for Parser and AddLoan

LS-Young committed:
edit DG

LS-Young committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

WeiXuanYap committed:
Add DG

WeiXuanYap committed:
Merge remote-tracking branch 'upstream/master'

jyxhazcake committed:
Merge pull request #237 from ZenoNexus/master

Add DG

YEOWEIHNGWHYELAB committed:
edit testing

YEOWEIHNGWHYELAB committed:
Merge pull request #238 from YEOWEIHNGWHYELAB/master

edit testing

YEOWEIHNGWHYELAB committed:
PPP

YEOWEIHNGWHYELAB committed:
Merge pull request #239 from YEOWEIHNGWHYELAB/master

PPP

YEOWEIHNGWHYELAB committed:
find update

YEOWEIHNGWHYELAB committed:
Merge pull request #241 from YEOWEIHNGWHYELAB/master

find update

EdisonZhong17 committed:
Fix minor bugs for delete feature

WeiXuanYap committed:
Add Doc Styling + Fix Help Command

WeiXuanYap committed:
Fix checkstyle

WeiXuanYap committed:
Merge pull request #242 from EdisonZhong17/master

Fix minor bugs for delete feature

WeiXuanYap committed:
Fix checkstyle issues

jyxhazcake committed:
Merge pull request #243 from ZenoNexus/master

Add Doc Styling + Fix Help Command

LS-Young committed:
update PPP

LS-Young committed:
merge

LS-Young committed:
Add comment for the list commands in TextUi

LS-Young committed:
edit comment

WeiXuanYap committed:
Fix add budget bug

1. When user first loads the storage, he is able to edit the budget of a month using `add` command. (Fixed)

WeiXuanYap committed:
fix checkstyle issues.

YEOWEIHNGWHYELAB committed:
refactor

YEOWEIHNGWHYELAB committed:
Merge pull request #245 from ZenoNexus/master

Fix add budget bug

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

# Conflicts:
#	src/main/java/seedu/budgettracker/logic/commands/AddBudgetCommand.java

YEOWEIHNGWHYELAB committed:
update ppp

YEOWEIHNGWHYELAB committed:
format

YEOWEIHNGWHYELAB committed:
Merge pull request #246 from YEOWEIHNGWHYELAB/master

refactor

LS-Young committed:
merge

YEOWEIHNGWHYELAB committed:
javadoc

EdisonZhong17 committed:
Merge pull request #244 from LS-Young/master

update PPP

YEOWEIHNGWHYELAB committed:
javadoc

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

YEOWEIHNGWHYELAB committed:
Merge pull request #247 from YEOWEIHNGWHYELAB/master

javadoc

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

EdisonZhong17 committed:
Fix boundary bugs for delete feature and update UG

EdisonZhong17 committed:
Merge remote-tracking branch 'origin/master'

EdisonZhong17 committed:
Fix boundary bugs for delete feature and update UG

EdisonZhong17 committed:
Merge pull request #248 from EdisonZhong17/master

Fix boundary bugs for delete feature and update UG

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

YEOWEIHNGWHYELAB committed:
puml ammendment

YEOWEIHNGWHYELAB committed:
Merge pull request #249 from YEOWEIHNGWHYELAB/master

PUML

YEOWEIHNGWHYELAB committed:
puml change

YEOWEIHNGWHYELAB committed:
update puml

YEOWEIHNGWHYELAB committed:
csv refactor

jyxhazcake committed:
Fixed bugs and add javadocs

jyxhazcake committed:
Merge pull request #250 from YEOWEIHNGWHYELAB/master

puml

YEOWEIHNGWHYELAB committed:
refactor csv

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

YEOWEIHNGWHYELAB committed:
Merge pull request #251 from YEOWEIHNGWHYELAB/master

refactor csv

jyxhazcake committed:
Merge branch 'master' into bug_fixing

# Conflicts:
#	src/main/java/seedu/budgettracker/data/AllRecordList.java
#	src/main/java/seedu/budgettracker/data/RecordList.java
#	src/main/java/seedu/budgettracker/logic/commands/AddBudgetCommand.java
#	src/main/java/seedu/budgettracker/logic/commands/EditBudgetCommand.java
#	src/main/java/seedu/budgettracker/logic/commands/EditExpenditureCommand.java
#	src/main/java/seedu/budgettracker/logic/parser/Parser.java

WeiXuanYap committed:
Merge pull request #252 from jyxhazcake/bug_fixing

Fixed bugs and added javaDocs

YEOWEIHNGWHYELAB committed:
convertToCsvFile puml

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

YEOWEIHNGWHYELAB committed:
Merge pull request #253 from YEOWEIHNGWHYELAB/master

puml

YEOWEIHNGWHYELAB committed:
reload images

YEOWEIHNGWHYELAB committed:
Merge pull request #254 from YEOWEIHNGWHYELAB/master

reload images

YEOWEIHNGWHYELAB committed:
ug summary

YEOWEIHNGWHYELAB committed:
Merge pull request #255 from YEOWEIHNGWHYELAB/master

ug summary

YEOWEIHNGWHYELAB committed:
edit to DG UG

jyxhazcake committed:
update UG

YEOWEIHNGWHYELAB committed:
dev guide check

jyxhazcake committed:
Merge pull request #256 from YEOWEIHNGWHYELAB/master

Doc

WeiXuanYap committed:
Auto stash before merge of "master" and "upstream/master"

WeiXuanYap committed:
Fix Checkstyle

jyxhazcake committed:
Merge pull request #257 from jyxhazcake/UserGuide

Update UG

jyxhazcake committed:
Merge pull request #258 from ZenoNexus/master

Add Developer Guide + Fix bugs

YEOWEIHNGWHYELAB committed:
doc

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

# Conflicts:
#	src/main/java/seedu/budgettracker/ui/TextUi.java

YEOWEIHNGWHYELAB committed:
format

YEOWEIHNGWHYELAB committed:
Merge pull request #259 from YEOWEIHNGWHYELAB/master

doc

WeiXuanYap committed:
Add DG - AddSequenceDiagram

jyxhazcake committed:
Update PPP and UG

jyxhazcake committed:
Merge pull request #260 from ZenoNexus/master

Add DG - AddSequenceDiagram

YEOWEIHNGWHYELAB committed:
ug

jyxhazcake committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp into UserGuide

# Conflicts:
#	docs/AboutUs.md

jyxhazcake committed:
Merge pull request #261 from YEOWEIHNGWHYELAB/master

ug

YEOWEIHNGWHYELAB committed:
ug

YEOWEIHNGWHYELAB committed:
update

YEOWEIHNGWHYELAB committed:
Merge pull request #263 from YEOWEIHNGWHYELAB/master

ug

YEOWEIHNGWHYELAB committed:
about us

WeiXuanYap committed:
Merge branch 'master' into UserGuide

WeiXuanYap committed:
Merge pull request #262 from jyxhazcake/UserGuide

Update AboutUs and PPP

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

# Conflicts:
#	docs/AboutUs.md

jyxhazcake committed:
Update DG for Implementation

jyxhazcake committed:
Merge pull request #265 from jyxhazcake/DeveloperGuide

Update DG for Implementation

jyxhazcake committed:
Update PPP and DG

jyxhazcake committed:
Merge pull request #266 from jyxhazcake/DeveloperGuide

Update PPP and DG

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

# Conflicts:
#	docs/AboutUs.md

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

WeiXuanYap committed:
Fix authorship tags

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

LS-Young committed:
Merge pull request #267 from ZenoNexus/master

Fix authorship tags

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

EdisonZhong17 committed:
Fix authorship tags

jyxhazcake committed:
Add end tag

jyxhazcake committed:
Add end tag

jyxhazcake committed:
Update RecordList.java

jyxhazcake committed:
Add end tag to Loan.java

jyxhazcake committed:
Add end authorship tag on HelpCommand

jyxhazcake committed:
Add end tags for TextUI.java and fix some tags

jyxhazcake committed:
Merge pull request #268 from EdisonZhong17/master

Fix authorship tags

LS-Young committed:
Add authortag

EdisonZhong17 committed:
Merge pull request #269 from LS-Young/master

Add authortag

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

EdisonZhong17 committed:
Update UG only

WeiXuanYap committed:
Change authorship tags

jyxhazcake committed:
Merge pull request #270 from EdisonZhong17/master

Update UG only

jyxhazcake committed:
Merge pull request #271 from ZenoNexus/master

Change authorship tags

YEOWEIHNGWHYELAB committed:
Update help, db, exit titles, faq, quick start

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

YEOWEIHNGWHYELAB committed:
bat file change

LS-Young committed:
Merge pull request #272 from YEOWEIHNGWHYELAB/master

Update Titles, FAQ, Quick Start

EdisonZhong17 committed:
Update UG with all links clickable and clean the command section format, update FAQ

YEOWEIHNGWHYELAB committed:
Merge pull request #273 from EdisonZhong17/master

Update UG with all links clickable and clean the command section form…

EdisonZhong17 committed:
Reformat the expected outcomes

EdisonZhong17 committed:
Reformat the expected outcomes and adding "back to top" links

WeiXuanYap committed:
Update UserGuide.md

WeiXuanYap committed:
Merge pull request #274 from EdisonZhong17/master

Reformat the expected outcomes and adding "back to top" links

jyxhazcake committed:
Add html code and Budgie

YEOWEIHNGWHYELAB committed:
update quick-start

jyxhazcake committed:
Change code

jyxhazcake committed:
Added meta charset

YEOWEIHNGWHYELAB committed:
update stat, bye, db, csv commands

WeiXuanYap committed:
Update UserGuide.md

jyxhazcake committed:
Revert to markdown code

jyxhazcake committed:
Test emoji

jyxhazcake committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp into UserGuide

# Conflicts:
#	docs/UserGuide.md

jyxhazcake committed:
Merge pull request #275 from YEOWEIHNGWHYELAB/master

update stat, bye, db, csv commands

WeiXuanYap committed:
Merge pull request #276 from jyxhazcake/UserGuide

Update User Guide with Notes (doesn't work yet)

WeiXuanYap committed:
Fix emoji + Add qns to FAQ + Add reader-centric content

jyxhazcake committed:
Merge pull request #277 from ZenoNexus/master

Fix emoji + Add qns to FAQ + Add reader-centric content

LS-Young committed:
Update UG with glossary

YEOWEIHNGWHYELAB committed:
fix emoji

jyxhazcake committed:
Add Introduction, Product Overview and Budgie in edit section

WeiXuanYap committed:
Merge pull request #278 from LS-Young/master

Update UG with glossary

WeiXuanYap committed:
Merge pull request #279 from YEOWEIHNGWHYELAB/master

fix emoji

jyxhazcake committed:
Update language

LS-Young committed:
Merge pull request #280 from jyxhazcake/UserGuide

Add Introduction, Product Overview and Budgie in edit section

YEOWEIHNGWHYELAB committed:
update formatting

YEOWEIHNGWHYELAB committed:
format

WeiXuanYap committed:
Update UserGuide.md

YEOWEIHNGWHYELAB committed:
update formatting

YEOWEIHNGWHYELAB committed:
Merge branch 'master' of https://github.com/YEOWEIHNGWHYELAB/tp

# Conflicts:
#	docs/UserGuide.md

YEOWEIHNGWHYELAB committed:
update description

WeiXuanYap committed:
Merge pull request #281 from YEOWEIHNGWHYELAB/master

update formatting

LS-Young committed:
Update emoji in list in UG

WeiXuanYap committed:
Add identified tag table + Edit TOC

WeiXuanYap committed:
Merge pull request #282 from LS-Young/master

Update emoji in list in UG

EdisonZhong17 committed:
Merge pull request #283 from ZenoNexus/master

Add identified tag table + Edit TOC

EdisonZhong17 committed:
Add emoji to delete part and reformatting

jyxhazcake committed:
Reformatted entire UG

jyxhazcake committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-F11-2/tp into UserGuide

# Conflicts:
#	docs/UserGuide.md

jyxhazcake committed:
Update delete position

jyxhazcake committed:
Fix typo

EdisonZhong17 committed:
Merge pull request #284 from jyxhazcake/UserGuide

Reformat User Guide

EdisonZhong17 committed:
Add emoji to delete part and page breaks

EdisonZhong17 committed:
Add emoji to delete part and page breaks

EdisonZhong17 committed:
Add emoji to delete part and page breaks

EdisonZhong17 committed:
Add emoji to delete part and page breaks

EdisonZhong17 committed:
Merge branch 'master' into master

jyxhazcake committed:
Merge pull request #285 from EdisonZhong17/master

Add emoji to delete part and page breaks
